### ISSUE TITLE
Bug: Executable zipped during Terraform hook loses execute permissions when extracted back into build artifact directory

### ISSUE THREAD

**Issue by ceptoplex** [aws/aws-sam-cli#4605]
### Description:

I'm using the new Terraform hook beta feature of SAM CLI for a project with Lambda function handlers written in Go.
The Terraform HCL configuration includes a `null_resource` to build the Go function handlers into executables which are then packaged into a .zip file.
For Go, these executables need to have the executable file permission bit or flag set before being archived into the .zip file.
When the `copy_terraform_built_artifacts.py` [extracts the .zip file](https://github.com/aws/aws-sam-cli/blob/62b37bdeeea8fc3191adffc57858ce72c59f10c6/samcli/hook_packages/terraform/copy_terraform_built_artifacts.py#L256) back into the SAM CLI build artifact directory (which defaults to `.aws-sam/build/...`), the extracted files lose there executable permission.

It seems [that this is the expected behavior of Python's `zipfile.extractall(...)`](https://stackoverflow.com/questions/39296101/python-zipfile-removes-execute-permissions-from-binaries) method?

### Steps to reproduce:

1. `git clone https://github.com/ceptoplex/aws-sam-cli-unzip-bug-poc`
2. `cd aws-sam-cli-unzip-bug-poc`
3. `cat README.md` will show all steps necessary to reproduce (it's not much!)

### Observed result:

```
~$ sam local invoke --beta-features --hook-name terraform -e events/example.json

Experimental features are enabled for this session.
Visit the docs page to learn more about the AWS Beta terms https://aws.amazon.com/service-terms/.

Skipped prepare hook. Current application is already prepared.
Invoking example (go1.x)
Skip pulling image and use local one: public.ecr.aws/sam/emulation-go1.x:rapid-1.70.0-x86_64.

Mounting /home/***/aws-sam-cli-zip-extraction-poc/.aws-sam/build/AwsLambdaFunctionThis*** as /var/task:ro,delegated inside runtime container
START RequestId: *** Version: $LATEST
fork/exec /var/task/example: permission denied: PathError
null
END RequestId: ***
REPORT RequestId: ***	Init Duration: 0.10 ms	Duration: 5.28 ms	Billed Duration: 6 ms	Memory Size: 128 MB	Max Memory Used: 128 MB	
{"errorMessage":"fork/exec /var/task/example: permission denied","errorType":"PathError"}
```

### Expected result:

```
~$ sam local invoke --beta-features --hook-name terraform -e events/example.json

Experimental features are enabled for this session.
Visit the docs page to learn more about the AWS Beta terms https://aws.amazon.com/service-terms/.

Skipped prepare hook. Current application is already prepared.
Invoking example (go1.x)
Skip pulling image and use local one: public.ecr.aws/sam/emulation-go1.x:rapid-1.70.0-x86_64.

Mounting /home/***/aws-sam-cli-zip-extraction-poc/.aws-sam/build/AwsLambdaFunctionThis*** as /var/task:ro,delegated inside runtime container
START RequestId: *** Version: $LATEST
{"statusCode":200,"headers":null,"multiValueHeaders":null,"body":"OK"}END RequestId: ***
REPORT RequestId: ***	Init Duration: 0.13 ms	Duration: 9.15 ms	Billed Duration: 10 ms	Memory Size: 128 MB	Max Memory Used: 128 MB
```

### Additional environment details

1. OS: Linux Mint 21.1 (vera) x64
4. `sam --version`: 1.70.0
5. AWS region: (any)

```
{
  "version": "1.70.0",
  "system": {
    "python": "3.7.10",
    "os": "Linux-5.15.0-58-generic-x86_64-with-debian-bookworm-sid"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.23",
    "aws_cdk": "Not available",
    "terraform": "1.3.7"
  }
}
```

**Comment by moelasmar** [comment]
thanks @ceptoplex for raising this issue. We will investigate it.

**Comment by moelasmar** [comment]
I can replicate this issue using the latest sam cli version.

**Comment by qingchm** [comment]
Hey guys this has been addressed in our most recent release 1.79.0 (https://github.com/aws/aws-sam-cli/releases/tag/v1.79.0). Closing the issue here. Thanks for opening the issue!
